[gomock generator] External mock bug if outside the app repository

```
gomock_generator --mocks-filename $SCALINGO_GOPATH/link/mocks.json --signatures-filename $SCALINGO_GOPATH/link/mocks_sig.json --debug
DEBU[2018-10-25T15:38:56.174] Execute mockgen command                       cmd="mockgen -destination /home/emichon/Documents/Scalingo/golang/src/github.com/Scalingo/link/scheduler/scheduler
mock/scheduler_mock.go  -package schedulermock github.com/Scalingo/link/scheduler Scheduler && sed -i s,github.com/Scalingo/link/vendor,, /home/emichon/Documents/Scalingo/golang/src/github.c
om/Scalingo/link/scheduler/schedulermock/scheduler_mock.go && goimports -w /home/emichon/Documents/Scalingo/golang/src/github.com/Scalingo/link/scheduler/schedulermock/scheduler_mock.go" dst
_package=schedulermock interface=Scheduler mock_file=scheduler/schedulermock/scheduler_mock.go nb_mocks=10                                                                                   
# github.com/coreos/etcd/clientv3
../client.go:336:9: undefined: metadata.NewContext
../lease.go:352:14: undefined: metadata.FromContext                                                                                                                                          
# github.com/coreos/etcd/clientv3                                                                                                                                                            
../client.go:336:9: undefined: metadata.NewContext
../lease.go:352:14: undefined: metadata.FromContext
# github.com/coreos/etcd/clientv3
../../../../coreos/etcd/clientv3/client.go:336:9: undefined: metadata.NewContext                                                                                                             
../../../../coreos/etcd/clientv3/lease.go:352:14: undefined: metadata.FromContext                                                                                                            
# github.com/coreos/etcd/clientv3
../../../../coreos/etcd/clientv3/client.go:336:9: undefined: metadata.NewContext
../../../../coreos/etcd/clientv3/lease.go:352:14: undefined: metadata.FromContext
# github.com/coreos/etcd/clientv3
/home/emichon/Documents/Scalingo/golang/src/github.com/coreos/etcd/clientv3/client.go:336:9: undefined: metadata.NewContext
/home/emichon/Documents/Scalingo/golang/src/github.com/coreos/etcd/clientv3/lease.go:352:14: undefined: metadata.FromContext
2018/10/25 15:38:59 Loading input failed: exit status 2
```

--- a/gomock_generator/gomockgenerator/generate.go
+++ b/gomock_generator/gomockgenerator/generate.go
@@ -20,8 +20,8 @@ import (
 
 // GenerationConfiguration lets you configure the generation of mocks for your project
 type GenerationConfiguration struct {
-	// MocksFilename is the filename of the JSON file containing the mock configuration.  Location of this file is the base package.
-	MocksFilename string
+	// MocksFilePath is the path to the JSON file containing the mock configuration.  Location of this file is the base package.
+	MocksFilePath string
 	// SignaturesFilename is the filename of the signatures cache. Location of this file is the base package.
 	SignaturesFilename string
 	// ConcurrentGoroutines specifies the concurrent amount of goroutines which can execute

--- a/gomock_generator/gomockgenerator/generate.go
+++ b/gomock_generator/gomockgenerator/generate.go
@@ -57,6 +57,15 @@ func GenerateMocks(ctx context.Context, gcfg GenerationConfiguration, mocksCfg M
 	if mocksCfg.BasePackage == "" {
 		panic(errors.New("BasePackage is mandatory"))
 	}
+	cwd, err := os.Getwd()
+	if err != nil {
+		return errors.Wrap(err, "fail to get current directory")
+	}
+	err = os.Chdir(path.Join(os.Getenv("GOPATH"), "src", mocksCfg.BasePackage))
+	if err != nil {
+		return errors.Wrap(err, "fail to move to base package directory")
+	}
+	defer os.Chdir(cwd)
 	log := logger.Get(ctx).WithField("nb_mocks", len(mocksCfg.Mocks))
 	ctx = logger.ToCtx(ctx, log)
 	log.WithFields(logrus.Fields{

--- a/gomock_generator/gomockgenerator/package.go
+++ b/gomock_generator/gomockgenerator/package.go
@@ -95,6 +95,10 @@ func fieldToString(field ast.Expr) string {
 		return fmt.Sprintf("[]%v", fieldToString(arrayExpr.Elt))
 	}
 
+	if ellipsisExpr, ok := field.(*ast.Ellipsis); ok {
+		return fmt.Sprintf("...%v", fieldToString(ellipsisExpr.Elt))
+	}
+
 	if _, ok := field.(*ast.InterfaceType); ok {
 		// TODO: It's the only case I can think of, but they might be others.
 		return "interface{}"

--- a/gomock_generator/main.go
+++ b/gomock_generator/main.go
@@ -39,7 +39,7 @@ Reads the mymocks.json file from the current directory and generates the mocks, 
 	app.cli.Usage = "Highly parallelized generator of gomock mocks"
 	app.cli.Version = "0.1.0"
 	app.cli.Flags = []cli.Flag{
-		cli.StringFlag{Name: "mocks-filename", Value: "mocks.json", Usage: "Filename of the JSON file containing the MockConfiguration. Location of this file is the base package.", EnvVar: "MOCKS_FILENAME"},
+		cli.StringFlag{Name: "mocks-filepath", Value: "./mocks.json", Usage: "Path to the JSON file containing the MockConfiguration. Location of this file is the base package.", EnvVar: "MOCKS_FILEPATH"},
 		cli.StringFlag{Name: "signatures-filename", Value: "mocks_sig.json", Usage: "Filename of the signatures cache. Location of this file is the base package.", EnvVar: "SIGNATURES_FILENAME"},
 		cli.IntFlag{Name: "concurrent-goroutines", Value: 4, Usage: "Concurrent amount of goroutines to generate mock.", EnvVar: "CONCURRENT_GOROUTINES"},
 		cli.BoolFlag{Name: "debug", Usage: "Activate debug logs"},
@@ -65,7 +65,7 @@ VERSION:
    {{end}}
 `
 	app.cli.Before = func(c *cli.Context) error {
-		app.config.MocksFilename = c.GlobalString("mocks-filename")
+		app.config.MocksFilePath = c.GlobalString("mocks-filepath")
 		app.config.SignaturesFilename = c.GlobalString("signatures-filename")
 		app.config.ConcurrentGoroutines = c.GlobalInt("concurrent-goroutines")
 		if c.GlobalBool("debug") {
@@ -76,12 +76,12 @@ VERSION:
 	}
 	app.cli.Action = func(_ctx *cli.Context) error {
 		log.WithFields(logrus.Fields{
-			"mocks_filename":        app.config.MocksFilename,
+			"mocks_file_path":       app.config.MocksFilePath,
 			"signatures_filename":   app.config.SignaturesFilename,
 			"concurrent_goroutines": app.config.ConcurrentGoroutines,
 		}).Info("Configuration for this mocks generation")
 
-		rawFile, err := os.Open(app.config.MocksFilename)
+		rawFile, err := os.Open(app.config.MocksFilePath)
 		if err != nil {
 			return errors.Wrap(err, "fail to open the mocks file")
 		}

